src/lbaf/Applications/LBAF_app.py

```diff
@@ -94,20 +94,22 @@ def parse_conf_file(self):
             if param_key in self.__allowed_config_keys:
                 self.__dict__[param_key] = param_val
 
-        # Parsing exodus if present
-        if self.configuration.get("exodus") is not None:
+        # Parse whether meshes must be generated
+        if (gm := self.configuration.get("generate_meshes")) is not None:
             self.grid_size = []
-            for key in ("x_procs", "y_procs", "z_procs"):
-                self.grid_size.append(self.configuration.get("exodus").get(key))
-            if math.prod(self.grid_size) != self.n_ranks:
-                raise AssertionError(f"Grid size: {self.grid_size} from exodus is not equal to n_ranks: {self.n_ranks}")
+            for key in ("x_ranks", "y_ranks", "z_ranks"):
+                self.grid_size.append(gm.get(key))
+            if math.prod(self.grid_size) < self.n_ranks:
+                self.logger.error(f"Grid size: {self.grid_size} < {self.n_ranks}")
+                sys.exit(1)
```

Comment:
As spoken, that is not a correct way to raise an error, it can not be tested. What was wrong with previous approach?

Reply:
`error message + exiting` can not be tested in unit test
Raising Error/Exception is exactly the same, but it can be tested